cheatsheets_draft/GraphQL_Cheat_Sheet.md

```diff
@@ -0,0 +1,128 @@
+
+#### Introspection
+
+Many implementations of GraphQL have Introspection enabled by default and leave it accessible to any incoming requests without requiring authentication. This is usually problematic because introspection allows the requester to learn all about supported schema and queries (see a [real-world example](https://hackerone.com/reports/291531) abusing this). Introspection might be how the API owner wants to educate consumers about how to use the API. However, the preferred way to educate consumers about a service is through a separate documentation channel such as a wiki, Git Readme, or readthedocs.
+
+The safest and usually easiest approach is to just disable introspection system-wide. See [this page](https://lab.wallarm.com/why-and-how-to-disable-introspection-query-for-graphql-apis/) or consult your GraphQL implementation's documentation to learn how to disable introspection altogether. If your implementation does not natively support disabling introspection or if you would like to allow some consumers/roles to have this access you can build a filter in your service to only allow approved consumers to access the introspection system.
+
+### IDOR Protection
+
+See the [IDOR prevention Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.html) for a comprehensive rundown of preventing IDOR.
+
+More specific to GraphQL, it's likely that you want consumers to be able to provide the direct identifier rather than creating an abstraction layer, which can be fine. In this case you would simply add a check that the requestor is able to access the ID that they gave. For APIs where individuals make their own requests (such as traditional website with backend API) you would just be checking that the ID given is owned by the user who is authenticated (e.g. does user 123 own picture with ID 564?). For more of a service account scenario where one consumer makes requests on behalf of many users it may not make sense to add a check if a given service account should be able to access any object.
+
+> The additional mitigation technique below probably belongs in the IDOR prevention cheat sheet but isn't there so I'll leave it here for now.
+
+One alternative protection would be creating an abstraction of the ID that is usually mapped to the requester's session. For example, instead of the user requesting a picture via its direct ID, they would request picture 2 in their account and the backend would use that abstract ID with the user's ID to derive the actual direct ID of the picture. This prevents the user from being able to provide their own direct identifier for an object and ensures that the user can only access objects belonging to them. Another abstraction mechanism could involve using different IDs for the frontend vs the backend with some unpredictable translation that happens on the backend where the user does not have any influence.
+
+## Other Resources
```

Comment:
Awesome resources - can we add some kind of categories to them?

Reply:
Yeah we can definitely categorize these so they're more easily consumable. I usually don't see sections like this in other sheets.

I'm thinking it might be cleaner to not include links in this section that were mentioned/linked previously in the doc, but I could go either way. Any thoughts on that?

Reply:
I would try to fit them along the CS where they make sense: I am not sure someone will go through a long list of links to read one by one. The remaining articles may be left as "Other Resources" as you have now.

Reply:
I removed any links that were already included earlier in the doc and split the remaining links into categories as well prettified the links.

Reply:
Gonna resolve this since I've implemented the feedback from this thread.